src/Common/src/Common.Hosting/HostBuilderExtensions.cs

```diff
@@ -104,28 +62,22 @@ private static IWebHostBuilder BindToPorts(this IWebHostBuilder webHostBuilder,
         {
             AddFromKubernetesEnv(urls);
         }
-        else
-        {
-            AddRunLocalPorts(urls, runLocalHttpPort, runLocalHttpsPort);
-        }
-
-        if (urls.Any())
-        {
-            // setting ASPNETCORE_URLS should only be needed to override launchSettings.json
-            if (string.IsNullOrWhiteSpace(portStr) && !Platform.IsKubernetes)
-            {
-                Environment.SetEnvironmentVariable("ASPNETCORE_URLS", string.Join(";", urls));
-            }
 
-            webHostBuilder.UseUrls(urls.ToArray());
-        }
-        else
+        if (!urls.Any())
         {
-            Environment.SetEnvironmentVariable("ASPNETCORE_URLS", DefaultUrl);
-            webHostBuilder.UseUrls(DefaultUrl);
+            urls.Add(DefaultUrl);
         }
 
-        return webHostBuilder;
+        return webHostBuilder.BindToPorts(urls);
+    }
+
+    private static IWebHostBuilder BindToPorts(this IWebHostBuilder webHostBuilder, List<string> urls)
+    {
+        string currentSetting = webHostBuilder.GetSetting(WebHostDefaults.ServerUrlsKey);
+        List<string> currentUrls = currentSetting?.Split(';').ToList() ?? new List<string>();
+        currentUrls.AddRange(urls);
+        IEnumerable<string> distinctUrls = currentUrls.Distinct();
+        return webHostBuilder.UseSetting(WebHostDefaults.ServerUrlsKey, string.Join(";", distinctUrls));
```

Comment:
Not sure if it matters much here, but if using a `HashSet<>` instead of `List<>`, there's no need to call `.Distinct()` which copies the collection.

Reply:
Bart, yes that makes sense - I made that change.